File: examples/python/connect.py
#!/usr/bin/env python3
"""Connect to Quantova and read chain state (QCore.py).

The simplest use of the Quantova tools, point the client at a gateway and read the
chain. No keys, no signing.

Run:
    export QUANTOVA_GATEWAY=http://127.0.0.1:8645   # or https://testnet.quantova.io
    python connect.py
"""

from _common import banner, connect, to_qtov

ADDRESS = "Q1GZD3AGFY5U426V9NX6UNE06ZC4YVKNK3GU9L3C"  # any Q1 address to inspect


def main():
    banner("Quantova: connect and read (QCore.py)")
    q = connect()

    info = q.node_info()
    print(f"Node {info.get('version', 'quantovad')} on chain {info.get('chain_id', 'Q-test-net')}")

    head = q.head()
    print(f"Head height: {head['height']}")

    account = q.get_account(ADDRESS)
    print(f"Balance of {ADDRESS}: {to_qtov(account.get('balance', 0))} QTOV  nonce {account.get('nonce', 0)}")


if __name__ == "__main__":
    main()

File: examples/javascript/connect.js
#!/usr/bin/env node
/**
 * Connect to Quantova and read chain state (QCore.js).
 *
 * Run:
 *   export QUANTOVA_GATEWAY=http://127.0.0.1:8645   // or https://testnet.quantova.io
 *   node connect.js
 *
 * In a full app with QCore.js installed:
 *   const { QCore } = require('@qunatovainc/qcore');
 *   const q = new QCore(process.env.QUANTOVA_GATEWAY);
 *   const head = await q.head();
 *   const account = await q.getAccount(ADDRESS);
 */
'use strict';
const { gw, toQtov, banner } = require('./_common');

const ADDRESS = 'Q1GZD3AGFY5U426V9NX6UNE06ZC4YVKNK3GU9L3C';

async function main() {
  banner('Quantova: connect and read (QCore.js)');

  const info = await gw('node_info');
  console.log(`Node ${info.version || info.node || 'quantovad'} on chain ${info.chain_id || info.chain || 'Q-test-net'}`);

  const head = await gw('head');
  console.log(`Head height: ${head.height}`);

  const account = await gw('get_account', { address: ADDRESS });
  console.log(`Balance of ${ADDRESS}: ${toQtov(account.balance || 0)} QTOV  nonce ${account.nonce || 0}`);
}
main().catch((e) => { console.error('error:', e.message); process.exit(1); });
